### PR TITLE
feedbackd: support cross-compilation

### PR DESCRIPTION
**Pull request by Mindavi**
Disables introspection for now.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux (cross-compiled)
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mindavi** [comment]
So yeah, this all seems correct, but the wrapGAppsHook produces a binary for x86-64... :disappointed:. See https://github.com/NixOS/nixpkgs/issues/175045. Guess we can still merge this anyway, but another fix is needed to actually make it fully work.

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/applications/misc/feedbackd/default.nix [hunk 1]
```diff
@@ -61,7 +62,11 @@ stdenv.mkDerivation rec {
     libgudev
   ];
 
-  mesonFlags = [ "-Dgtk_doc=true" "-Dman=true" ];
+  mesonFlags = [
+    "-Dgtk_doc=${lib.boolToString (stdenv.buildPlatform == stdenv.hostPlatform)}"
+    "-Dman=true"
+    "-Dintrospection=${if (stdenv.buildPlatform == stdenv.hostPlatform) then "enabled" else "disabled"}"
```
[the remark this answers is not in the document]
Comment (Mindavi) [reply pkgs/applications/misc/feedbackd/default.nix:68]: I'll add a link here: https://github.com/NixOS/nixpkgs/issues/72868

[the remark this answers is not in the document]
Comment (Mindavi) [reply pkgs/applications/misc/feedbackd/default.nix:70]: I am honestly not completely sure.

File: pkgs/applications/misc/feedbackd/default.nix [hunk 2]
```diff
@@ -32,7 +32,8 @@ stdenv.mkDerivation rec {
   # only a Debian package release that is tagged in the upstream repo
   version = "0.0.0+git20220520";
 
-  outputs = [ "out" "dev" "devdoc" ];
+  outputs = [ "out" "dev" ]
+    ++ lib.optionals (stdenv.buildPlatform == stdenv.hostPlatform) [ "devdoc" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/feedbackd/default.nix:36]:
```suggestion
    # remove if cross-compiling gobject-introspection works
    ++ lib.optionals (stdenv.buildPlatform == stdenv.hostPlatform) [ "devdoc" ];
```

File: pkgs/applications/misc/feedbackd/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/applications/misc/feedbackd/default.nix:70]: I did a quick readup. We are basically missing the GObject bindings to other languages.
What does that mean in practical terms? For example you couldn't load them in python anymore. I am not sure what could break there but I imagine that maybe you cannot load the library there or parse all DBus messages.

So sounds to me like that programs in languages other than C that use Gnome related things will break and GUI programs written in Python could break in unexpected ways.

https://source.puri.sm/Librem5/feedbackd/-/blob/master/libfeedback/meson.build#L83-118

https://source.puri.sm/Librem5/feedbackd/-/blob/master/doc/overview.xml#L28-30

TLDR: We should fix this rather than adding this everywhere.

@jtojnar did I misunderstand something?

[the remark this answers is not in the document]
Comment (jtojnar) [reply pkgs/applications/misc/feedbackd/default.nix:70]: Yeah, without introspection, the library cannot be used from languages like Python or JavaScript (well there is always manual FFI – but hopefully no project is masochistic enough to do that). It should fail with an error like the following at runtime so hopefully people will notice:

```Python traceback
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/nix/store/4lrl4lnlkigfb6jnv3p7qix337yz6zya-python3-3.9.12-env/lib/python3.9/site-packages/gi/importer.py", line 136, in load_module
    raise ImportError('cannot import name %s, '
ImportError: cannot import name Lfb, introspection typelib not found
```

Fixing it properly is hard, see #72868.

**Comment by Mindavi** [comment]
To me it wouldn't make sense that the option of disabling introspection would be available if it broke _everything_, but that may be naive.

I agree that keeping simple would be nice. I don't agree that it aligns with upstream if they provide the option.

Some progress is being made on improving the gir situation! https://github.com/NixOS/nixpkgs/pull/176464

So we may be able to remove the disables in the near future.

Personally I like to be able to cross-compile, even if it reduces functionality a bit. I do try it out from time to time and am aiming to get a full-blown image (with DM) working at some point, just because I like to prove it's possible and to be able to use my own hardware to build the image. But of course it would be nice to have more functionality.

**Review by tomfitzhenry — APPROVED** [review]

File: pkgs/applications/misc/feedbackd/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mindavi) [reply pkgs/applications/misc/feedbackd/default.nix:71]: Yeah, kinda depends on which application needs to use this and how they do it.
